schema-www/src/pages/[item].tsx

```diff
@@ -0,0 +1,144 @@
+  }));
+
+  return { paths, fallback: false };
+};
+
+export const getStaticProps: GetStaticProps<{}, Params> = async ({
+  params,
+}) => {
+  if (!params) throw new Error("Missing params");
+
+  const filePath = path.join(process.cwd(), "../rdfs", `regen.ttl`);
+  const fileContents = await fs.readFile(filePath, "utf8");
+
+  // Parse the RDF data
+  const store = rdf.graph();
+  rdf.parse(fileContents, store, BASE_URI, "text/turtle");
+  const item = REGEN(`${params.item}`);
+
+  const itemType = store.any(item, RDF("type"));
+  if (!itemType) throw new Error(`No type found for ${item}`);
+
+  if (itemType.value == RDF("Property").value) {
+    // Fetch property data
+    const label = store.any(item, RDFS("label"))?.value;
+    const comment = store.any(item, RDFS("comment"))?.value;
+    const ranges = store
+      .statementsMatching(item, RDFS("range"), undefined)
+      .map((statement) => statement.object.value);
+    const domains = store
+      .statementsMatching(item, RDFS("domain"), undefined)
+      .map((statement) => statement.object.value);
+
+    return {
+      props: {
+        item: {
+          type: "property",
+          iri: item.value,
+          label,
+          description: comment,
+          ranges,
+          domains,
+        },
+      },
+    };
+  } else if (itemType.value == RDFS("Class").value) {
+    // Fetch class data
+    const label = store.any(item, RDFS("label"))?.value;
+    const comment = store.any(item, RDFS("comment"))?.value;
+    const properties = store
+      .statementsMatching(undefined, RDFS("domain"), item)
+      .map((statement) => {
+        const label = statement.subject.value;
+        const ranges = store
+          .statementsMatching(statement.subject, RDFS("range"), undefined)
+          .map((propRange) => {
+            return propRange.object.value;
+          });
+        const description = store.any(
+          statement.subject,
+          RDFS("comment")
+        )?.value;
+        return { type: "property", label, ranges, description };
+      });
+
+    return {
+      props: {
+        item: {
+          type: "class",
+          iri: item.value,
+          label,
+          description: comment,
+          properties,
+        },
+      },
+    };
+  } else {
+    throw new Error(`Unhandled type for ${item}`);
+  }
+};
+
+type ItemProps = {
+  item: Property | Class;
+};
+
+const SchemaPage: React.FC<ItemProps> = ({ item }) => {
+  if (item.type === "class") {
+    return <ClassView {...item} />;
+  } else if (item.type === "property") {
+    return <PropertyView {...item} />;
+  } else {
+    return (
+      <div className={styles.error}>
+        <h1 className={styles.title}>Error: Invalid schema type</h1>
+        <p className={styles.text}>
+          The provided schema type is not recognized. Please ensure it is either
+          &quot;class&quot; or &quot;property&quot;.
+        </p>
+      </div>
+    );
+  }
```

Comment:
It's more like a convention but conditionally render item is often in JSX directly to improve readability:

```suggestion
 const isClass = item.type === "class";
 const isProperty = item.type === "property";
 const isDefault = !isClass & !isProperty
 
    return (
    <>
     {isClass && <ClassView {...item} />;}
     {isItem && <PropertyView {...item} />;}
     {isDefault && <div className={styles.error}>
        <h1 className={styles.title}>Error: Invalid schema type</h1>
        <p className={styles.text}>
          The provided schema type is not recognized. Please ensure it is either
          &quot;class&quot; or &quot;property&quot;.
        </p>
      </div>}
     </>
    );
  }
```

Reply:
I actually don't prefer the approach @flagrede outlines here just because it isn't clearly conditional what renders (IE at a glance, it's not clear if two of those conditions can be met at the same time), but it's a bit more clear if you return early on the conditional cases and avoid `else` blocks:

```tsx
  if (item.type === "class") {
    return <ClassView {...item} />;
  } 
  if (item.type === "property") {
    return <PropertyView {...item} />;
  }
  return (
    <div className={styles.error}>
      <h1 className={styles.title}>Error: Invalid schema type</h1>
      <p className={styles.text}>
        The provided schema type is not recognized. Please ensure it is either
        &quot;class&quot; or &quot;property&quot;.
      </p>
    </div>
  );
}
```

Reply:
Interesting, I didn't know we had different thoughts on this topic within the team. Regarding the point you outline @haveanicedavid if a component is meant to render only one of its children at a time then using a state that is unique instead of variables for each block clears the confusion. That way you have the same clarity while having only a single return for the component.

Reply:
Is this what you meant @flagrede ? https://github.com/regen-network/regen-registry-standards/pull/54/commits/7a65b0365943073ad00d6d016a660eef04f2c13f

Reply:
Not exactly, this is what I meant: https://github.com/regen-network/regen-registry-standards/pull/54/files#r1226226721
Whenever possible I try to have only one return statement by component. Here since the condition to return is based on a unique state you can be sure that you're never going to have `isClass` and `isProperty` at the same time. In case your condition is more complex (it depends on several variables) you can still use that pattern by having a util function that returns a unique state (ie: `const myState = getComponent(item.type, item.value, item.name)`. Does that make sense?

Reply:
> Whenever possible I try to have only one return statement by component
I actually prefer multiple return statements vs having conditionals live within one JSX block because I find it easier to read / understand. Not trying to argue with you here btw, I think the suggested changes you made are perfectly fine, just wanted to explain my reasoning:

If I see several `&&` statements within one JSX block, I have to check whatever those conditions are and mentally grok whether or not they could happen at the same time, conflict with each other, etc. If I see multiple returns statements, I can read them sequentially knowing that if the first condition is met, the rest won't render -- to me, it's just a bit simpler.

Like many of my code pattern opinions, it isn't really something i'm passionate about, just a preference. As long as things are easy to read (which everything you offered in feedback are!), I don't mind which approach is taken

tldr: the suggestions @flagrede is making sound good to me (even if I prefer something a bit different)